docs/dotnet/dotnet-network-traces.md

```diff
@@ -0,0 +1,462 @@
+overloaded, the rate of connection-related spans is expected to be much lower than the rate of
+HTTP client request spans.
+
+Applications are encouraged to enable *HTTP client request* spans by default in production environments.
+
+Connection-level spans are experimental - their semantics may be changed in the future in a breaking manner.
+Using connection-level instrumentation in production environments should be done after appropriate validation.
+
+The connection-related spans are reported only by [HttpClientHandler](https://learn.microsoft.com/dotnet/api/system.net.http.httpclienthandler)
+and [SocketsHttpHandler](https://learn.microsoft.com/dotnet/api/system.net.http.socketshttphandler) which may not be supported
+on certain platforms or may not be used by a particular application.
+
+## HTTP client request
+
+**Status**: [Stable][DocumentStatus]
+
+.NET `HttpClient` reports client request spans according to [HTTP Client Semantic Conventions](/docs/http/http-spans.md#http-client-span) with the following
+specific details:
+
+- `network.protocol.name`, `network.peer.port`, and `http.request.resend_count` attributes are not reported
+- `url.full` is redacted by default - query parameter values are replaced with `*`. Redaction can be disabled by setting `AppContext` switch `System.Net.Http.DisableQueryRedaction` to `true`.
+- When the `error.type` attribute is reported, it contains one of [HTTP Request errors](https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror) in snake_case, a full exception type name, or a string representation of received status code.
+- All attributes are reported after `Activity` is started, none are provided at creation time.
+- In case redirects occur, each redirected request is reported as a separate span.
+- `SocketsHttpHandler` may retry requests on connection failure. Such retries are not reported as separate spans.
+
+Corresponding `Activity.OperationName` is `System.Net.Http.HttpRequestOut`, `ActivitySource` name - `System.Net.Http`.
+Span with HTTP semantics was added in .NET 9.
+
+## HTTP client request: wait for connection
+
+<!-- semconv span.dotnet.http.request.wait_for_connection.internal -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+**Status:** ![Development](https://img.shields.io/badge/-development-blue)
+
+The span describes the time it takes for the HTTP request to obtain a connection from the connection pool.
+
+The span is reported only if there was no connection readily available when request has started.
+It's reported as a child of *HTTP client request* span.
+
+The span ends when the connection is obtained - it could happen when an existing connection becomes available or once
+a new connection is established, so the duration of *Wait For Connection* span is different from duration of the
```

Comment:
let's try to generalize it and try to address similar problems - concurrency, mutexes, etc, look at absl